### PR TITLE
Add lead history component from detailed lead page

### PR DESCRIPTION
**Pull request by rudenkoserhii**
## Describe your changes

- I created component LeadHistory, which collect information about lead's Activity from Qobrix .
- Implemented ability clear all filters on property leads
- Deleted unneccessaries old files
- Made component Button close
- Fixed the same height of the properties row.

P.S. Too many changes, because of merging with PR75 (they are both very connected and it was very uncomfortrable work without merging. So if it's hard to check it, You con check it after aproving PR75 and merging it to develop.

Screenshot 2023-12-25 233528

Screenshot 2023-12-25 232850

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-179?atlOrigin=eyJpIjoiN2QxOTZlNjZjNGU2NGYzZTk0NGFhZTZjYzFiMTUxY2MiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/app/[locale]/global.css [hunk 1]
```diff
@@ -24,16 +24,26 @@ img {
...
 }
 
 /* End Normalize */
 
 .slick-dots {
   top: 10rem;
   transform: scale(2);
-  filter: invert(1);
+  /* filter: invert(1); */
   height: fit-content;
```
Comment (AKutusenko) [inline src/app/[locale]/global.css]: commented code

File: src/app/[locale]/global.css [hunk 2]
```diff
@@ -24,16 +24,26 @@ img {
...
+  & > li > button::before {
+    opacity: 0.5 !important;
+    color: #00a76f !important;
+    text-shadow: 0px 0px 2px white;
```
Comment (AKutusenko) [inline src/app/[locale]/global.css]: color shoud be in const

File: src/app/[locale]/global.css [hunk 3]
```diff
@@ -24,16 +24,26 @@ img {
...
+  }
+  & > .slick-active > button::before {
+    opacity: 1 !important;
+    color: #007867 !important;
+  }
 }
```
Comment (AKutusenko) [inline src/app/[locale]/global.css]: same

File: src/app/[locale]/property/[id]/[...leads]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+'use client';
+
+import { ProtectedRoute } from 'components/custom';
+
+import { Common } from '../../../leads/lib/components/common/common';
+
```
Comment (AKutusenko) [inline src/app/[locale]/property/[id]/[...leads]/page.tsx]: imports should be absolute

File: src/assets/illustrations/page-not-matched-illustration.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import { memo } from 'react';
+// @mui
+import { useTheme } from '@mui/material/styles';
+import Box, { BoxProps } from '@mui/material/Box';
+//
+import BackgroundShape from './background-shape';
+
+// ----------------------------------------------------------------------
+
```
Comment (AKutusenko) [inline src/assets/illustrations/page-not-matched-illustration.tsx:9]: comment

File: src/assets/illustrations/page-not-matched-illustration.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        fill="url(#paint0_linear_1_119)"
+        d="M175 265.6c1-8.7-12.1-4.8-17-5.6v-66.6c0-4.5-1.5-5.6-5.6-5.6-5.3.3-13.8-1.4-17.1 4l-55 68.3c-2.7 3.3-1.8 8.8-2 12.8 0 4.1 1.5 5.6 5.6 5.6h54.7v21.7c-.9 7.9 9.1 5.2 13.7 5.6 4.1 0 5.6-1.5 5.6-5.6v-21.7c13.8-1.1 18.1 4.5 17.1-12.9zm-72.5-5.6l36-44.4V260h-36zm309.1 5.6c1-8.7-12.2-4.8-17.1-5.6v-66.6c0-4.5-1.5-5.6-5.6-5.6-5.3.3-13.7-1.4-17.1 4l-55 68.3c-2.7 3.3-1.9 8.8-2 12.8 0 4.1 1.5 5.6 5.6 5.6h54.7v21.7c-.9 7.9 9.1 5.2 13.7 5.6 4.1 0 5.6-1.5 5.6-5.6v-21.7c14.1-1.1 18.2 4.5 17.2-12.9zm-72.4-5.6l36-44.4V260h-36z"
+      />
+
+      <path
+        fill={PRIMARY_MAIN}
+        d="M425.6 118.2c0-5-4.6-9-9.6-8.2-2-3.7-6-6-10.2-5.9 4.3-21.4-30-21.4-25.7 0-8.7-.8-15.1 9.4-10.4 16.8 2.1 3.5 5.9 5.6 10 5.5h38.7v-.1c4.1-.4 7.2-3.9 7.2-8.1zM104.3 200c.1-4.2-4.1-7.8-8.2-7-1.7-3.2-5.1-5.1-8.8-5 3.8-18.4-25.8-18.4-22 0-7.4-.7-12.9 8.1-8.9 14.4 1.8 3 5.1 4.8 8.6 4.7h33.2v-.1c3.4-.4 6.1-3.4 6.1-7z"
+        opacity="0.08"
+      /> */}
+
```
Comment (AKutusenko) [inline src/assets/illustrations/page-not-matched-illustration.tsx]: commented code

File: src/components/custom/leadsList/components/lead-item/lead-item.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+import { useRouter } from 'next/navigation';
+
+import { useTranslations } from 'next-intl';
+
+import { Box, Card } from '@mui/material';
+import { AppRoute } from 'enums';
+import { QobrixLeadItem } from 'types';
+
+import { TextStyled } from '../../styles';
+
```
Comment (AKutusenko) [inline src/components/custom/leadsList/components/lead-item/lead-item.tsx]: imports should be absolute

File: src/components/custom/leadsList/index.tsx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                  bottom: '-0.5rem',
+                  right: '-0.5rem',
+                  backgroundColor: colors.PRIMARY_LIGHT_COLOR,
+                  transition: 'all 200ms ease-out',
+                  '&:hover': {
+                    color: colors.BUTTON_SECOND_COLOR,
+                    transition: 'all 200ms ease-out',
+                  },
+                }}
+              /> */}
```
Comment (AKutusenko) [inline src/components/custom/leadsList/index.tsx]: commented code

File: src/components/custom/property/hooks/useMagnifyingGlass.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+const useMagnifyingGlass = ({
+  magnifierHeight,
+  magnifierWidth,
+  zoomLevel,
+}: UseMagnifyingGlassOptions) => {
+  const [[x, y], setXY] = useState<[number, number]>([0, 0]);
+  const [[imgWidth, imgHeight], setSize] = useState<[number, number]>([0, 0]);
+  const [showMagnifier, setShowMagnifier] = useState(false);
+
+  const handleMouseEnter = (x: number, y: number, width: number, height: number) => {
```
Comment (AKutusenko) [inline src/components/custom/property/hooks/useMagnifyingGlass.ts]: returning params type

Comment (serhii-rudenko) [reply src/components/custom/property/hooks/useMagnifyingGlass.ts:18]: fixed

File: src/components/custom/leadsList/index.tsx (hunk 8 above)
Comment (serhii-rudenko) [reply src/components/custom/leadsList/index.tsx:99]: fixed

File: src/components/custom/leadsList/components/lead-item/lead-item.tsx (hunk 7 above)
Comment (serhii-rudenko) [reply src/components/custom/leadsList/components/lead-item/lead-item.tsx:12]: fixed

File: src/assets/illustrations/page-not-matched-illustration.tsx (hunk 6 above)
Comment (serhii-rudenko) [reply src/assets/illustrations/page-not-matched-illustration.tsx:56]: fixed

File: src/assets/illustrations/page-not-matched-illustration.tsx (hunk 5 above)
Comment (serhii-rudenko) [reply src/assets/illustrations/page-not-matched-illustration.tsx:9]: fixed

File: src/app/[locale]/property/[id]/[...leads]/page.tsx (hunk 4 above)
Comment (serhii-rudenko) [reply src/app/[locale]/property/[id]/[...leads]/page.tsx:6]: fixed

File: src/app/[locale]/global.css (hunk 3 above)
Comment (serhii-rudenko) [reply src/app/[locale]/global.css:47]: fixed

File: src/app/[locale]/global.css (hunk 2 above)
Comment (serhii-rudenko) [reply src/app/[locale]/global.css:41]: fixed

File: src/app/[locale]/global.css (hunk 1 above)
Comment (serhii-rudenko) [reply src/app/[locale]/global.css:37]: fixed

File: src/components/custom/property/hooks/useMagnifyingGlass.ts (hunk 9 above)
Comment (rudenkoserhii) [reply src/components/custom/property/hooks/useMagnifyingGlass.ts]: fixed

File: src/components/custom/leadsList/index.tsx (hunk 8 above)
Comment (rudenkoserhii) [reply src/components/custom/leadsList/index.tsx]: fixed

File: src/components/custom/leadsList/components/lead-item/lead-item.tsx (hunk 7 above)
Comment (rudenkoserhii) [reply src/components/custom/leadsList/components/lead-item/lead-item.tsx]: fixed

File: src/assets/illustrations/page-not-matched-illustration.tsx (hunk 6 above)
Comment (rudenkoserhii) [reply src/assets/illustrations/page-not-matched-illustration.tsx]: fixed

File: src/assets/illustrations/page-not-matched-illustration.tsx (hunk 5 above)
Comment (rudenkoserhii) [reply src/assets/illustrations/page-not-matched-illustration.tsx:9]: fixed

File: src/app/[locale]/property/[id]/[...leads]/page.tsx (hunk 4 above)
Comment (rudenkoserhii) [reply src/app/[locale]/property/[id]/[...leads]/page.tsx]: fixed

File: src/app/[locale]/global.css (hunk 3 above)
Comment (rudenkoserhii) [reply src/app/[locale]/global.css]: fixed

File: src/app/[locale]/global.css (hunk 2 above)
Comment (rudenkoserhii) [reply src/app/[locale]/global.css]: fixed

File: src/app/[locale]/global.css (hunk 1 above)
Comment (rudenkoserhii) [reply src/app/[locale]/global.css]: fixed

**Review by AKutusenko — APPROVED** [review]

**Pull request merged**